src/tools/adig.c

```diff
@@ -151,15 +155,88 @@ static void print_help(void)
     "              SOA, SRV, TXT, TLSA, URI, CAA, SVCB, HTTPS\n\n");
 }
 
+static int split_rcline(char *rcline, char **rcargv, int len) {
+  int   rcargc;
+  char *last = NULL;
+
+  for (rcargc = 1, rcargv[rcargc] = strtok_r(rcline, " \n", &last);
+    rcargc < len && rcargv[rcargc];
+    rcargv[++rcargc] = strtok_r(NULL, " \n", &last))
+  {
+    DEBUGF(fprintf(stderr, "split_rcline() argc: %d, argv: %s\n", rcargc, rcargv[rcargc]));
+  }
+
+  return (rcargc);
+}
+
 static ares_bool_t read_cmdline(int argc, const char * const *argv,
-                                adig_config_t *config)
+                                adig_config_t *config, ares_bool_t is_rcfile)
 {
-  ares_getopt_state_t state;
-  int                 c;
+  ares_getopt_state_t  state;
+  int                  c;
+  char                 configdir[256];
+  char                *configdir_xdg;
+  unsigned int         cdlen = 0;
+  unsigned int         cdsize = sizeof(configdir);
+  char                *homedir;
 
   ares_getopt_init(&state, argc, argv);
   state.opterr = 0;
 
+  if (!is_rcfile) {
+
+#if defined(WIN32)
+    cdlen = (unsigned int) snprintf(configdir, cdsize, "%s", getenv("%%APPDATA%"));
+
```

Comment:
should just be APPDATA here, no %, the % is just the syntax Windows uses for environment variables in the old cmd.exe shell.  Also need to specify a sub-directory here I'd think like you do for macos.

Reply:
Thanks, will adjust.